src/main/java/io/temporal/internal/statemachines/ActivityStateMachine.java

```diff
@@ -0,0 +1,404 @@
+        .add(
+            State.STARTED_ACTIVITY_CANCEL_COMMAND_CREATED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_COMPLETED,
+            State.COMPLETED,
+            ActivityStateMachine::cancelCommandNotifyCompleted)
+        .add(
+            State.STARTED_ACTIVITY_CANCEL_COMMAND_CREATED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_FAILED,
+            State.FAILED,
+            ActivityStateMachine::cancelCommandNotifyFailed)
+        .add(
+            State.STARTED_ACTIVITY_CANCEL_COMMAND_CREATED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_TIMED_OUT,
+            State.TIMED_OUT,
+            ActivityStateMachine::cancelCommandNotifyTimedOut)
+        .add(
+            State.STARTED_ACTIVITY_CANCEL_EVENT_RECORDED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_FAILED,
+            State.FAILED,
+            ActivityStateMachine::notifyFailed)
+        .add(
+            State.STARTED_ACTIVITY_CANCEL_EVENT_RECORDED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_COMPLETED,
+            State.COMPLETED,
+            ActivityStateMachine::notifyCompleted)
+        .add(
+            State.STARTED_ACTIVITY_CANCEL_EVENT_RECORDED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_CANCELED,
+            State.CANCELED,
+            ActivityStateMachine::notifyCancellationFromEvent);
+  }
+
+  private final ExecuteActivityParameters parameters;
+
+  private final Functions.Proc2<Optional<Payloads>, Failure> completionCallback;
+
+  private long startedCommandEventId;
+
+  /**
+   * @param parameters attributes used to schedule an activity
+   * @param completionCallback one of ActivityTaskCompletedEvent, ActivityTaskFailedEvent,
+   *     ActivityTaskTimedOutEvent, ActivityTaskCanceledEvents
+   * @param commandSink sink to send commands
+   * @return an instance of ActivityCommands
+   */
+  public static ActivityStateMachine newInstance(
+      ExecuteActivityParameters parameters,
+      Functions.Proc2<Optional<Payloads>, Failure> completionCallback,
+      Functions.Proc1<CancellableCommand> commandSink) {
+    return new ActivityStateMachine(parameters, completionCallback, commandSink);
+  }
+
+  private ActivityStateMachine(
+      ExecuteActivityParameters parameters,
+      Functions.Proc2<Optional<Payloads>, Failure> completionCallback,
+      Functions.Proc1<CancellableCommand> commandSink) {
+    super(newStateMachine(), commandSink);
+    this.parameters = parameters;
+    this.completionCallback = completionCallback;
+    explicitEvent(ExplicitEvent.SCHEDULE);
+  }
+
+  public void createScheduleActivityTaskCommand() {
+    addCommand(
+        Command.newBuilder()
+            .setCommandType(CommandType.COMMAND_TYPE_SCHEDULE_ACTIVITY_TASK)
+            .setScheduleActivityTaskCommandAttributes(parameters.getAttributes())
+            .build());
+  }
+
+  public void cancel() {
+    if (parameters.getCancellationType() == ActivityCancellationType.ABANDON) {
```

Comment:
(more for my own knowledge than an actual comment), what does "Abandon" mean vs some of the other semantics and why would a developer choose one over the other

Reply:
Let me know if the [existing JavaDoc ](https://github.com/temporalio/java-sdk/blob/e9361fc01840a9e26e49effe0c52240f23583f94/src/main/java/io/temporal/activity/ActivityCancellationType.java#L30)is not clear.